src/templates/plugin.main.js

```diff
@@ -136,7 +136,9 @@ export default async (context) => {
 
     let matchedLocale
 
-    if (useCookie && (matchedLocale = app.i18n.getLocaleCookie())) {
+    if (strategy !== STRATEGIES.NO_PREFIX && (matchedLocale = getLocaleFromRoute(route)) && route.path !== '/') {
+      // Get preferred language from locale route
+    } else if (useCookie && (matchedLocale = app.i18n.getLocaleCookie())) {
```

Comment:
I'm not even sure if the `doDetectBrowserLanguage` function is the right place to handle this. There is some logic before `doDetectBrowserLanguage` is called that determines locale to use based on other factors and settings and this change will potentially override that previous decision.

I will have to graph all possible flows in my head and try to test them to figure out what's the best place to handle that.

Reply:
@rchl did you get a chance to take a look if not I might try to something again.

Thanks!

Reply:
I've started making tests. I will put out a PR so you can track the progres.
I'm on vacation next two weeks so won't have the time to work on it probably.

Reply:
Hi @rchl, I do really hope that you'd a wonderful vacation and felling good!

I'm just patiently waiting for that PR, maybe there is something that I could be helpful, please just let me know, thanks!

Reply:
Sorry for the long wait. I'll try to address it as soon as I can.
I've decided to implement it as an option and potentially make it a default in the future as I'm not confident that it won't break some use cases.